rclpy/rclpy/node.py

```diff
@@ -14,24 +14,30 @@
 
 import rclpy
 
+from rclpy.client import Client
 from rclpy.publisher import Publisher
-from rclpy.subscription import Subscription
 from rclpy.qos import qos_profile_default
+from rclpy.service import Service
+from rclpy.subscription import Subscription
 
 
 class Node:
 
     def __init__(self, handle):
+        self.clients = []
         self.handle = handle
+        self.publishers = []
+        self.services = []
```

Comment:
The callback group has weak pointers to them, which is a slightly different thing than what is happening here in Python.